src/antsibull_core/utils/hashing.py

```diff
@@ -8,14 +8,32 @@
 from __future__ import annotations
 
 import hashlib
+import typing as t
+from collections.abc import Mapping
 
 import aiofiles
 
 from .. import app_context
 
 
+class _AlgorithmData(t.NamedTuple):
+    name: str
+    algorithm: str
+    kwargs: dict
```

Comment:
```suggestion
    kwargs: dict[str, t.Any]
```

Reply:
Implemented in c71d9d39b24d7c457c4a2160c2984734632f4160.